app/src/main/java/com/github/se/project/ui/lesson/SelectedTutorDetails.kt

```diff
@@ -0,0 +1,379 @@
+              IconButton(
+                  onClick = { navigationActions.goBack() },
+                  modifier = Modifier.testTag("backButton")) {
+                    Icon(
+                        imageVector = Icons.AutoMirrored.Filled.ArrowBack,
+                        contentDescription = "Back")
+                  }
+            },
+            title = {
+              Text(
+                  text = "Available Tutors",
+                  style = MaterialTheme.typography.titleLarge,
+                  modifier = Modifier.testTag("confirmLessonTitle"))
+            })
+      }) { paddingValues ->
+        Column(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(paddingValues)
+                    .padding(horizontal = 16.dp)
+                    .verticalScroll(rememberScrollState())
+                    .testTag("selectedTutorDetailsScreen"),
+            verticalArrangement = Arrangement.spacedBy(8.dp)) {
+
+              // Check if the given profile is a tutor
+              if (tutorProfile.role != Role.TUTOR) {
+                ErrorState(message = "No tutor selected. Should not happen.")
+              } else {
+                // Tutor information card
+                Card(
+                    modifier = Modifier.fillMaxWidth(),
+                    colors =
+                        CardDefaults.cardColors(
+                            containerColor = MaterialTheme.colorScheme.surfaceVariant)) {
+                      DisplayTutorDetails(tutorProfile, listProfilesViewModel)
+                    }
+
+                Spacer(modifier = Modifier.weight(1f))
+
+                // Confirmation button
+                Button(
+                    shape = MaterialTheme.shapes.medium,
+                    onClick = { showConfirmDialog = true },
+                    modifier =
+                        Modifier.fillMaxWidth().padding(bottom = 16.dp).testTag("confirmButton")) {
+                      Icon(
+                          Icons.AutoMirrored.Filled.Send,
+                          contentDescription = "Confirmation Button Icon",
+                          modifier = Modifier.size(20.dp))
+                      Spacer(modifier = Modifier.width(8.dp))
+                      Text("Confirm your lesson")
+                    }
+              }
+            }
+
+        // Confirmation Dialog
+        if (showConfirmDialog) {
+          AlertDialog(
+              modifier = Modifier.testTag("confirmDialog"),
+              onDismissRequest = { showConfirmDialog = false },
+              title = {
+                Text(
+                    text = "Confirm Your Choice", modifier = Modifier.testTag("confirmDialogTitle"))
+              },
+              text = {
+                Text(
+                    text =
+                        "Would you like to choose ${tutorProfile.firstName} ${tutorProfile.lastName} for your lesson and pay a price of ${tutorProfile.price}.-/hour?",
```

Comment:
The phrasing here is maybe a little awkward, "pay a price" might make people thinking they have to pay upfront